contracts/ERC20.sol

```diff
@@ -0,0 +1,131 @@
+pragma solidity ^0.5.16;
+
+import "./SafeMath.sol";
+
+contract ERC20 {
+    using SafeMath for uint;
+
+    string public name;
+    string public symbol;
+    uint8 public decimals;
+    uint  public totalSupply;
+    address public operator;
+    mapping(address => uint) public balanceOf;
+    mapping(address => mapping(address => uint)) public allowance;
+    mapping (address => bool) public minters;
+
+    bytes32 public DOMAIN_SEPARATOR;
+    // keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+    bytes32 public constant PERMIT_TYPEHASH = 0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
+    mapping(address => uint) public nonces;
+
+    event Approval(address indexed owner, address indexed spender, uint value);
+    event Transfer(address indexed from, address indexed to, uint value);
+    event AddMinter(address indexed minter);
+    event RemoveMinter(address indexed minter);
+    event ChangeOperator(address indexed newOperator);
+
+    modifier onlyOperator {
+        require(msg.sender == operator, "ONLY OPERATOR");
+        _;
+    }
+
+    constructor(string memory name_, string memory symbol_, uint8 decimals_) public {
+        name = name_;
+        symbol = symbol_;
+        decimals = decimals_;
+        operator = msg.sender;
+        uint chainId;
+        assembly {
+            chainId := chainid
+        }
+        DOMAIN_SEPARATOR = keccak256(
+            abi.encode(
+                keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)'),
+                keccak256(bytes(name)),
+                keccak256(bytes('1')),
+                chainId,
+                address(this)
+            )
+        );
+    }
+
+    function changeOperator(address operator_) public onlyOperator {
```

Comment:
No safer function for transfering here? Like `setPendingOperator` and `claimOperator`?

Reply:
will add

Reply:
done. Please resolve